traits/has_traits.py

```diff
@@ -1355,7 +1355,8 @@ def __dir__(self):
         trait_names = self.trait_names()
         method_names = [method for method in self._each_trait_method(self)]
         class_attrs = list(vars(self.__class__).keys())
-        return trait_names + method_names + class_attrs
+        return list(
+            set(trait_names + method_names + class_attrs + super().__dir__()))
```

Comment:
I _think_ if you use `super` then you shouldn't need to  do `method_names` and `class_attrs`: they should be picked up by the standard implementation.

Reply:
Yes, you are right, verified that. Thanks.